### PR TITLE
Component/secondary button

### LINKED ISSUE #7: SecondaryButton
Build SecondaryButton component.

Screenshot 2020-12-13 at 21 14 11

View the component in [Figma](https://www.figma.com/file/gGwSe7ilZCUmAIaJyUc4iY/Bristol-Tech-Fair-Design-System?node-id=489%3A58)

### PR DESCRIPTION
**Pull request by ammersive**

### CODE REVIEW

**Comment by ammersive** [comment]
First pass at secondary button styling, with a test for content.

**Review by RobTF9 — DISMISSED** [review]

File: src/components/SecondaryButton/SecondaryButton.styles.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  align-items: center;
+  background: var(--lightblue);
+  color: var(--lightgrey);
+  font-weight: 700;
+  padding: var(--smallest-space) var(--small-space);
+  border-radius: 6.25rem;
+  &:hover {
+    background: var(--logopink);
+    color: var(--lightgrey);
+  }
```
[the remark this answers is not in the document]
Comment (ammersive) [reply src/components/SecondaryButton/SecondaryButton.styles.js]: Great, have added medium space to padding and arranged other styling (hover and active) to be in line with primary button.

File: src/components/SecondaryButton/SecondaryButton.test.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import SecondaryButton from './SecondaryButton';
+
+describe('SecondaryButton', () => {
+  test('renders SecondaryButton component', () => {
+    render(<SecondaryButton content="Hello" />);
+
+    expect(screen.getByText('Hello'));
+  });
+});
```
[the remark this answers is not in the document]
Comment (ammersive) [reply src/components/SecondaryButton/SecondaryButton.test.jsx]: Thanks, test now added.

**Comment by ammersive** [comment]
Edits after initial review: padding change, active & focus styling in line with primary btn, and test for onClick added

File: src/components/SecondaryButton/SecondaryButton.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <SecondaryButtonWrapper type={type} onClick={onClick}>
+      {content}
+    </SecondaryButtonWrapper>
+  );
+};
+
+SecondaryButton.propTypes = {
+  content: PropTypes.string.isRequired,
+  type: PropTypes.oneOf(['button', 'submit', 'reset']),
+  onClick: PropTypes.func
```
Comment (shanesc) [inline src/components/SecondaryButton/SecondaryButton.jsx:16]: Just got to thinking, should we require the `onClick` prop? Not necessary but would avoid accidentally having buttons that don't do anything. I didn't do this on PrimaryButton but if you think it's worthwhile I'll update it.

What do you think?

**Review by shanesc — APPROVED** [review]
Looks great Sophie!

Agreed about the typography, we can update for all buttons once we add the bold font face.

I added one comment about adding `onClick: PropTypes.func.isRequired`, but if you don't think it's needed go ahead and hit squash and merge!

File: src/components/SecondaryButton/SecondaryButton.jsx (hunk 3 above)
Comment (RobTF9) [reply src/components/SecondaryButton/SecondaryButton.jsx:16]: No, you don't necessarily need one. In forms for example, the event is handled my the submission of the form as opposed to the button so a onClick is not always required.

**Review by RobTF9 — APPROVED** [review]
Ready to merge, click "Squash and merge" when you're ready

File: src/components/SecondaryButton/SecondaryButton.jsx (hunk 3 above)
Comment (ammersive) [reply src/components/SecondaryButton/SecondaryButton.jsx:16]: Thanks for taking a look Shane, and for flagging up whether we require onClick - and thanks for the clarity on that Rob. Makes sense to me!  I'll go ahead and squash and merge.

**Comment by ammersive** [comment]
Thanks for the review Rob and Shane. Squashing and merging now.

**Pull request merged**